docs/advanced-recipes/atom-creators.mdx

```diff
@@ -356,3 +356,66 @@ const ListOfThings = () => {
 ```
 
 <CodeSandbox id="ugkzm0" />
+
+## atomWithDebounce
+
+> `atomWithDebounce` helps with creating an atom where state set should be debounced.
+
+This can be especially useful for text search inputs, where you would like to fire an action only after the user has stopped typing for a delay, instead of firing an action on every keystroke.
+
+```tsx
+export function atomWithDebounce<Value>(
+  initialValue: Value,
+  delayMilliseconds = 500,
+  shouldDebounceOnReset = false
+) {
+  let timeout: NodeJS.Timeout
```

Comment:
Let's remove this for now as this shared variable is not correct for atom creators.

Reply:
Continuing from where we left off, could we set `timeout` to be an atom that stores the `timeoutId` then? Would this solve the issue?

Reply:
Do you mean this?
```ts
const timeoutAtom = atom<NodeJS.Timeout | null>(null)
```
Yea, it should solve the issue.

Reply:
Did amendments here: https://codesandbox.io/s/withered-tdd-cjrz2y?file=/src/atomWithDebounce.ts (will update the PR code at a later date after https://github.com/pmndrs/jotai/pull/1338#discussion_r942555474 pending discussion has been addressed)

I still do not see when collisions can occur as I thought the `window` object would guarantee that there will be no possible occurrence in the browser. If you have the time, could you perhaps share an instance where this might occur for my own learning?

Reply:
https://codesandbox.io/s/beautiful-shannon-gy153b?file=/src/App.tsx

It's not easy to see from the UI (because `isDebouncingAtom` is an atom correctly), but it behaves incorrectly.

Reply:
I see! just forked and tried it myself. You're right that it behaves incorrectly. Thanks for doing up this sandbox! TIL.